doc/manual/manual.tex

```diff
@@ -10530,6 +10530,20 @@ \section{Extending and contributing to \aspect}
 \url{https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions}
 contain a section on how to debug \dealii{} programs.
 
+\subsection{The main control flow and the \texttt{Simulator} class}
+\label{sec:simulator}
+
+In order to modify \aspect{} it is helpful to have a general understanding of the structure of the program. Even though you will most likely not need to change something in these high-level routines, it will give a better overview of how all the individual plugin systems (see Section \ref{sec:plugins}) work together. \aspect{}'s execution begins in \url{source/main.cc} in the function \texttt{main()}. This function has a number of initialization tasks (see Figure~\ref{fig:program-structure}), and finally creates an object of the class \href{https://aspect.geodynamics.org/doc/doxygen/classaspect_1_1Simulator.html}{aspect::Simulator}, and calls the function \texttt{Simulator::run()} that contains the main loop that advances the model in time (see Panel 3 in Figure~\ref{fig:program-structure}), until its termination criteria are fulfilled. You can imagine the \texttt{Simulator} class as \textbf{the} particular model you are running, it is meant to represent everything that encompasses your model, its matrices, solution vectors, geometries, material properties, and so on. All of this is created in the constructor of the \texttt{Simulator} class (Panel 2 of Figure \ref{fig:program-structure}). However, the \texttt{Simulator} class does not contain all of this functionality in itself (that would be overwhelming and bad software design). Instead it owns a number of other objects that do many sub-tasks, from setting boundary conditions, to assembling specific terms in the matrices, to prescribing the material properties. The main simulator object only orchestrates all of these member objects to create a succesful model run. Many of these sub-objects need to look very differently for different models, and therefore they are designed as plugin systems as described in the next section.
```

Comment:
A few stylistic suggestions:

`This function has a number of initialization tasks (see Figure~\ref{fig:program-structure}) and finally creates an object of the class \href{https://aspect.geodynamics.org/doc/doxygen/classaspect_1_1Simulator.html}{aspect::Simulator}. {aspect::Simulator} calls the function \texttt{Simulator::run()}, which contains the main loop that advances the model in time (see Panel 3 in Figure~\ref{fig:program-structure}) until its termination criteria are fulfilled`

`it is meant to` -> `as it is meant to`

`its matrices, solution vectors, geometries, material properties, and so on` -> `(matrices, solution vectors, geometries, material properties, and so on)`

Reply:
If you break this paragraph into individual lines of ~80 characters, it becomes easier to comment :-)

Either way: 
* "you will most likely not need to change something" -> "you will most likely not need to change anything"
* "an object of the class" -> "an object of class"
* this sentence has "and finally..., and" -- remove one "and", but the "finally" also appears wrong :-)
* no comma in front of "until"; I would move the parenthetical remark after the "until" part to make reading easier
* \textbf{the} -> \textem{the}